test/extended/image_ecosystem/mongodb_replica_statefulset.go

```diff
@@ -94,11 +95,21 @@ var _ = g.Describe("[image_ecosystem][mongodb][Slow] openshift mongodb replicati
 			err = oc.Run("delete").Args("pods", "--all", "-n", oc.Namespace()).Execute()
 			o.Expect(err).ShouldNot(o.HaveOccurred())
 
-			g.By("waiting for all pods to reach running status")
+			g.By("waiting for all pods to be gracefully deleted")
 			podNames, err = exutil.WaitForPods(
 				oc.KubeClient().Core().Pods(oc.Namespace()),
 				exutil.ParseLabelsOrDie("name=mongodb-replicaset"),
-				exutil.CheckPodIsRunningFn,
+				func(pod kapi.Pod) bool { return pod.DeletionTimestamp != nil },
```

Comment:
are we guaranteed to see the pod while it has a deletion timestamp? what if it "gracefully deletes" very quickly?